Track Buddy/MotionManager.swift

```diff
@@ -66,12 +69,57 @@ class MotionManager: ObservableObject {
     private var recentPoints: Deque<CGPoint> = [] // TODO: dluo- think about thread safety?
     
     func pointPath(atScale factor: CGFloat) -> CGMutablePath {
-        let points = recentPoints.map { CGPoint(x: $0.x * factor, y: $0.y * factor) }
+        let xPoints = recentPoints.map { Double($0.x * factor) }
+        let yPoints = recentPoints.map { Double($0.y * factor) }
+        
+        /* Interpolate points to fill in gaps between rate-limited points to create a smooth tracer line.
+         
+         Using this interpolate-to-smooth approach because generating the line with the raw accelerometer data at 100 Hz
+         can produce visible divergence between the movement of the dot in the graph and the line that is traced under it.
+         
+         */
+        
+        let interpolatedXPoints = interpolate(xPoints)
+        let interpolatedYPoints = interpolate(yPoints)
+        
         let path = CGMutablePath()
-        path.addLines(between: points)
+        guard interpolatedXPoints.count == interpolatedYPoints.count else { return path }
+        
+        var interpolatedPoints: [CGPoint] = []
+        for i in 0..<interpolatedXPoints.count {
+            interpolatedPoints.append(CGPoint(
+                x: interpolatedXPoints[i],
+                y: interpolatedYPoints[i]
+            ))
+        }
```

Comment:
Seems like there's some cleanup we could do to avoid this, but I'd have to play with it. Anywhere from restructuring the `interpolate` function to take in an array of CGPoints directly, or maybe using an intermediary dictionary (perhaps with index as key and (x,y) stored as tuples), a type extension somewhere, etc.

Reply:
Yeah I can definitely play with cleaning this up before merging